tests/series/arithmetic/test_floordiv.py

```diff
@@ -0,0 +1,253 @@
+    check(assert_type(left_i // i, pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i // f, pd.Series), pd.Series, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        assert_type(left_i // c, Never)
+        assert_type(left_i // s, Never)
+        assert_type(left_i // d, Never)
+
+    # `numpy` typing gives the corresponding `ndarray`s in the static type
+    # checking, where our `__rfloordiv__` cannot override. At runtime, they lead to
+    # errors or pd.Series.
+    check(b // left_i, pd.Series, np.integer)
+    check(i // left_i, pd.Series, np.integer)
+    check(f // left_i, pd.Series, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        assert_type(c // left_i, Any)
+        assert_type(s // left_i, Any)
+        assert_type(d // left_i, Any)  # pyright: ignore[reportAssertTypeFailure]
+
+    check(assert_type(left_i.floordiv(b), pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i.floordiv(i), pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i.floordiv(f), pd.Series), pd.Series, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        left_i.floordiv(c)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
+        left_i.floordiv(s)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
+        # left_i.floordiv(d)  # This invalid one cannot be detected by static type checking
+
+    check(assert_type(left_i.rfloordiv(b), pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i.rfloordiv(i), pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i.rfloordiv(f), pd.Series), pd.Series, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        left_i.rfloordiv(c)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
+        left_i.rfloordiv(s)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
+    check(
+        assert_type(left_i.rfloordiv(d), "pd.Series[pd.Timedelta]"),
+        pd.Series,
+        pd.Timedelta,
+    )
+
+
+def test_floordiv_pd_index(left_i: pd.Series) -> None:
+    """Test pd.Series[int] // pandas Indexes"""
+    b = pd.Index([True, True, True])
+    i = pd.Index([2, 3, 5])
+    f = pd.Index([1.0, 2.0, 3.0])
+    c = pd.Index([1.1j, 2.2j, 4.1j])
+    s = pd.Index([datetime(2025, 10, d) for d in (27, 28, 29)])
+    d = pd.Index([timedelta(seconds=s + 1) for s in range(3)])
+
+    check(assert_type(left_i // b, pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i // i, pd.Series), pd.Series, np.integer)
+    check(assert_type(left_i // f, pd.Series), pd.Series, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _03 = left_i // c  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _04 = left_i // s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        # _05 = left_i // d  # This invalid one cannot be detected by static type checking
```

Comment:
A bit surprising, but then I see that you didn't add `s` and `d` in the tests in `tests/series/arithmetic/test_truediv.py`, so you should add that and things should be the same between floordiv and truediv